setup.py

```diff
@@ -17,7 +17,7 @@
 ]
 extras_require = {
     "checking": ["black", "flake8", "isort", "mdformat"],
-    "optional": ["mlflow", "GitPython"],
+    "optional": ["mlflow", "GitPython", "scikit-learn", "torch", "transformers"],
```

Comment:
I think torch should be moved to "checking" because it is used in the tests.

Reply:
Yeah I wanted to chat about it. Should we perhaps add a "testing" option?

For dependencies specifically needed for unit tests. Whereas "checking" right now covers formatting dependencies.

Reply:
> Should we perhaps add a "testing" option?

Yes please.